pkg/transform/jq_test.go

```diff
@@ -52,6 +52,25 @@ func TestJQRunFunction_SpMode_true(t *testing.T) {
 			ExpInterState: nil,
 			Error:         nil,
 		},
+		{
+			Scenario:  "test_timestamp_to_epochMillis_context",
+			JQCommand: `{ sessionId: .contexts_com_snowplowanalytics_snowplow_client_session_1[0].firstEventTimestamp | epochMillis }`,
```

Comment:
So if it's a string, but no layout is provided, we simply default to the layout that we encounter in Snowplow data - is that correct?

Reply:
Yes. Also default layout in `epoch`  is different than the layout we use for date-time atomic fields:

* for atomic like `collector_tstamp` we use `yyyy-MM-dd HH:mm:ss.SSS`
* for contexts it's `yyyy-MM-dd'T'HH:mm:ss.SSS'Z'` (so `T` + `Z`), based on [tracker code](https://github.com/snowplow/snowplow-ios-tracker/blob/bf1495987d63dc3595270df1a1bb516bfef8585f/Sources/Core/Utils/Utilities.swift#L80). And based on what I've seen in generated mobile data and `firstEventTimestamp` field :)

Reply:
That's correct. For any Snowplow field, from JQ transformation point of view, in 99.99% cases you'd only need pure `epoch` or `epochMillis`, no custom layout needed. If there is some weird format suddenly coming from somewhere, you can add param to handle it.